autopilot/sagemaker_autopilot_direct_marketing.ipynb

```diff
@@ -0,0 +1,647 @@
+   "metadata": {},
+   "source": [
+    "We can use the result of the transform job to evaluate additional metrics on the test dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.metrics import (\n",
+    "    average_precision_score,\n",
+    "    classification_report,\n",
+    "    f1_score,\n",
+    "    precision_recall_curve,\n",
+    "    roc_auc_score,\n",
+    "    roc_curve,\n",
+    ")\n",
+    "\n",
+    "labels = test_data[\"y\"]\n",
+    "AUC = roc_auc_score(labels == \"yes\", test_data_preds.probability)\n",
+    "AP = average_precision_score(labels, test_data_preds.probability, pos_label=\"yes\")\n",
+    "F1 = f1_score(labels, test_data_preds.predicted_label, pos_label=\"yes\")\n",
+    "\n",
+    "print(f\"AUC: {AUC:.3f}\\nAP {AP:.3f}\\nF1 {F1:.3f}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(classification_report(labels == \"yes\", test_data_preds.predicted_label == \"yes\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "plt.figure(num=None, figsize=(16, 9), dpi=160, facecolor=\"w\", edgecolor=\"k\")\n",
+    "fpr, tpr, _ = roc_curve(labels == \"yes\", test_data_preds.probability)\n",
+    "plt.plot(fpr, tpr)\n",
+    "plt.plot([0, 1], [0, 1])\n",
+    "plt.xlabel(\"False Positive Rate\")\n",
+    "plt.ylabel(\"True Positive Rate\")\n",
+    "plt.title(\"ROC Curve\")\n",
+    "\n",
+    "\n",
+    "plt.figure(num=None, figsize=(16, 9), dpi=160, facecolor=\"w\", edgecolor=\"k\")\n",
+    "precision, recall, _ = precision_recall_curve(\n",
+    "    labels == \"yes\", test_data_preds.probability\n",
+    ")\n",
+    "plt.plot(recall, precision)\n",
+    "plt.plot([0, 1], [1, 0])\n",
+    "plt.xlabel(\"Recall\")\n",
+    "plt.ylabel(\"Precision\")\n",
+    "plt.title(\"Precision-Recall Curve\")\n",
+    "plt.title(\"ROC Curve\");"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Candidate Generation Notebook\n",
+    "    \n",
+    "Sagemaker AutoPilot also auto-generates a Candidate Definitions notebook. This notebook can be used to interactively step through the various steps taken by the Sagemaker Autopilot to arrive at the best candidate. This notebook can also be used to override various runtime parameters like parallelism, hardware used, algorithms explored, feature extraction scripts and more.\n",
+    "    \n",
+    "The notebook can be downloaded from the following Amazon S3 location:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "automl_job.describe_auto_ml_job()[\"AutoMLJobArtifacts\"][\n",
+    "    \"CandidateDefinitionNotebookLocation\"\n",
```

Comment:
Candidate definition notebooks contain hyperlinks to related files elsewhere in the Autopilot output dir (e.g. the `dpp-0.py` implementation) that don't work if you only download the notebook itself... Would be nice to mention this, and maybe demonstrate just `aws s3 sync`ing the entire Autopilot output folder to the notebook's data dir?

Reply:
Changed to download entire folder, and also included download of explainability report. This is particularly relevant in case of SSO auth, when user might not have access to AWS Console.

Reply:
I was actually shooting for even broader scope, to demonstrate that you even have access to the Python code implementing the pre-transformations both for batch and real-time (in case you wanted to customize further)... I think this is really useful for showing how you might be able to use Autopilot output together with human data science (or even deploy to edge in some cases).

Have just pushed a modified version closer to what I had in mind as part of the merge - but it does increase the initial download time quite a bit - have a look & would be great to hear what you think and if we can further change/revert it!

Reply:
I see what you mean now. Dumping the entire output folder will also copy the model artifacts of all the candidates. 
What about doing a full dump only for the best candidate? we can merge it with the Explainability piece, since this is the only candidate with explainability already computed.